core/domain/user_services.py

```diff
@@ -295,6 +296,10 @@ def set_username(user_id, new_username):
     user_settings.username = new_username
     _save_user_settings(user_settings)
 
+def set_signup_profile_picture_url(user_id):
```

Comment:
Call this generate_profile_picture_data_url

and add a docstring: Generates a profile picture data URL for a new user. This uses their gravatar, if it exists, otherwise it generates an identicon based on their email address.